dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/InactivityValidationAwareConnectionKeepAliveStrategy.java

```diff
@@ -86,6 +88,15 @@ public TimeValue getKeepAliveDuration(HttpResponse response, HttpContext context
                 }
             }
         }
+
+        ProtocolVersion responseVersion = response.getVersion();
+        if (responseVersion != null && responseVersion.lessEquals(HttpVersion.HTTP_1_0)) {
+            log.info(
+                    "HTTP/1.0 requires keep-alive negotiation, but no timeout was found",
+                    SafeArg.of("protocolVersion", responseVersion));
+            return TimeValue.ZERO_MILLISECONDS;
+        }
```

Comment:
I think this suggests that `ConnectionReuseStrategy.keepAlive` incorrectly returned `true` -- I think the expectation is for `getKeepAliveDuration` to be called after a `ConnectionReuseStrategy.keepAlive` check

Reply:
yeah, this case should already be handled by [DefaultConnectionReuseStrategy.keepAlive](https://github.com/apache/httpcomponents-core/blob/rel/v5.3.4/httpcore5/src/main/java/org/apache/hc/core5/http/impl/DefaultConnectionReuseStrategy.java#L107-L160) returning false, I'm trying to run down a reproducer to test with full HTTP headers.

Reply:
We may be able to adapt something like this to reproduce the observed behavior: #2538